src/python_testing/TC_OCC_2_2.py

```diff
@@ -0,0 +1,146 @@
+
+    def desc_TC_OCC_2_2(self) -> str:
+        return "[TC-OCC-2.2] OccupancySensorTypeBitmap and OccupancySensorType interdependency with server as DUT"
+
+    def steps_TC_OCC_2_2(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read OccupancySensorType attribute selection based on FeatureMap Bitmap."),
+            TestStep(3, "Read OccupancySensorTypeBitmap attribute selection based on FeatureMap Bitmap.")
+        ]
+        return steps
+
+    def pics_TC_OCC_2_2(self) -> list[str]:
+        pics = [
+            "OCC.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_OCC_2_2(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        attributes = Clusters.OccupancySensing.Attributes
+        feature_map = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.FeatureMap)
+        is_pir_feature_supported = feature_map & Clusters.OccupancySensing.Bitmaps.Feature.kPassiveInfrared
+        is_us_feature_supported = feature_map & Clusters.OccupancySensing.Bitmaps.Feature.kUltrasonic
+        is_phy_feature_supported = feature_map & Clusters.OccupancySensing.Bitmaps.Feature.kPhysicalContact
+
+        self.step(1)
+        attribute_list = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+
+        self.step(2)
+        # OccupancySensorType will be determined by FeatureMap matching table at 2.7.6.2.
+        if attributes.OccupancySensorType.attribute_id in attribute_list:
+            occupancy_sensor_type_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorType)
+
+            # No Bitmap support from PIR, US, Physical Contact, then OccupancySensorType should be PIR
+            if (is_pir_feature_supported != 1) & (is_us_feature_supported != 1) & (is_phy_feature_supported != 1):
+                asserts.assert_equal(occupancy_sensor_type_dut,
+                                     Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPir, "OccupancySensorType is not PIR")
+            # Bitmap supports PIR, then OccupancySensorType should be PIR
+            elif (is_pir_feature_supported == 1) & (is_us_feature_supported != 1) & (is_phy_feature_supported != 1):
+                asserts.assert_equal(occupancy_sensor_type_dut,
+                                     Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPir, "OccupancySensorType is not PIR")
+            # Bitmap supports US, then OccupancySensorType should be US
+            elif (is_pir_feature_supported != 1) & (is_us_feature_supported == 1) & (is_phy_feature_supported != 1):
+                asserts.assert_equal(
+                    occupancy_sensor_type_dut, Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kUltrasonci, "OccupancySensorType is not Ultrasonic")

```

Comment:
typo in `kUltrasonic` here

```suggestion
                    occupancy_sensor_type_dut, Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kUltrasonic, "OccupancySensorType is not Ultrasonic")

```